StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1129CodeFixProvider.cs

```diff
@@ -49,21 +50,116 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
+            }
+            else if (IsEnumWithDefaultMember(namedTypeSymbol, out memberName))
+            {
+                replacement = ConstructMemberAccessSyntax(newExpression.Type, memberName);
+            }
+            else
+            {
+                replacement = SyntaxFactory.DefaultExpression(newExpression.Type);
+            }
+
+            return replacement
                 .WithLeadingTrivia(newExpression.GetLeadingTrivia())
                 .WithTrailingTrivia(newExpression.GetTrailingTrivia());
         }
 
+        /// <summary>
+        /// Determines whether a symbol is an instance of a given <see cref="Type"/>.
+        /// </summary>
+        /// <typeparam name="T">The type to match.</typeparam>
+        /// <param name="namedTypeSymbol">The symbol.</param>
+        /// <returns><see langword="true"/> if the syntax matches the type; <see langword="false"/> otherwise.</returns>
+        private static bool IsType<T>(INamedTypeSymbol namedTypeSymbol)
+        {
+            if (namedTypeSymbol == null)
+            {
+                return false;
+            }
+
+            var expectedType = typeof(T);
+
+            if (!string.Equals(expectedType.Name, namedTypeSymbol.Name, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            if (!string.Equals(
+                expectedType.Namespace,
+                namedTypeSymbol.ContainingNamespace?.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted)),
+                StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether a given enumeration symbol contains a member with value <c>0</c>.
+        /// </summary>
+        /// <param name="namedTypeSymbol">The symbol.</param>
+        /// <param name="foundMemberName">Will be set to the string name of the member, if one is found.</param>
+        /// <returns><see langword="true"/> if the syntax is an enumeration with a value of <c>0</c>; <see langword="false"/> otherwise.</returns>
+        private static bool IsEnumWithDefaultMember(INamedTypeSymbol namedTypeSymbol, out string foundMemberName)
+        {
+            foundMemberName = null;
+
+            if (namedTypeSymbol == null || namedTypeSymbol.TypeKind != TypeKind.Enum)
+            {
+                return false;
+            }
+
+            var foundMembers = namedTypeSymbol
+                .GetMembers()
+                .Where(m => m.Kind == SymbolKind.Field)
+                .OfType<IFieldSymbol>()
+                .Where(fs => fs.ConstantValue.Equals(0))
+                .ToList();
+
+            if (foundMembers.Count != 1)
+            {
+                return false;
+            }
+
+            foundMemberName = foundMembers.Single().Name;
```

Comment:
:thought_balloon:  Since `foundMembers` is already a list, `foundMembers[0].Name` will work more efficiently.